### PR TITLE
Remove all useragent sniffing

### PR DESCRIPTION
**Pull request by oreoshake**
Fixes https://github.com/twitter/secureheaders/issues/384

I was trying to convert my app to use `frame-src` instead of `child-src` but found the UA sniffing getting in the way. No more magic around the way `child-src` and `frame-src` are handled.

This will result in more console errors but I'm not sure if it will actually cause any breakage. I think the long chain of fallback rules will save us here.

### CODE REVIEW

**Comment by oreoshake** [comment]
With some light testing this resulted in one console warning in safari (`worker-src`) and one less warning on firefox (discouraging child-src in favor of `frame/worker-src`).

File: docs/upgrading-to-6-0.md [hunk 1]
```diff
@@ -48,3 +48,7 @@ Prior to 6.0.0 you could conceivably, though unlikely, have `Configure#default`
...
 
 The first example: we will now send `'unsafe-inline'` along with nonce source expressions. This will generate warnings in some consoles but is 100% valid use and was a design goal of CSP in the early days. The concept of versioning CSP lost out and so we're left with backward compatibility as our only option.
+
+## No more frame-src/child-src magic
+
+First there was frame-src. Then there was child-src which deprecated frame-src. Then child-src became deprecated in favor of frame-src and worker-src. In the meantime, every browser did something different. For a while, it was recommended to set child-src and frame-src but the values had to be identical. secure_headers would sniff the UA to determine whether to use frame or child src. That can lead to confusing things like setting frame-src but seeing child-src. If the child-src and frame-src did not match up, an error is raised. This can be very confusing when using dynamic overrides ("Do we use child-src or frame-src?" => :boom:). Now that the dust has settled, I think we can stop sniffing UAs and just go with a straightforward application.
```
Comment (jacobbednarz) [inline docs/upgrading-to-6-0.md]: :tada:

File: lib/secure_headers/headers/content_security_policy.rb [hunk 2]
```diff
@@ -219,13 +185,13 @@ def append_nonce(source_list, nonce)
       source_list
     end
 
-    # Private: return the list of directives that are supported by the user agent,
+    # Private: return the list of directives,
```
Comment (jacobbednarz) [inline lib/secure_headers/headers/content_security_policy.rb]: Do you want to rewrap this line?

**Comment by jacobbednarz** [comment]
Looks reaaaaal nice 💯

File: lib/secure_headers/headers/content_security_policy.rb (hunk 2 above)
Comment (oreoshake) [reply lib/secure_headers/headers/content_security_policy.rb]: Heh, I should probably take a pass over all of the comments. I'm sure there's plenty of outdated info.

File: lib/secure_headers/headers/policy_management.rb [hunk 3]
```diff
@@ -199,9 +153,9 @@ module ClassMethods
       #
       # Returns a default policy if no configuration is provided, or a
       # header name and value based on the config.
-      def make_header(config, user_agent = nil)
+      def make_header(config)
```
Comment (oreoshake) [inline lib/secure_headers/headers/policy_management.rb]: Removing the `user_agent` argument is a breaking change that will require a deprecation release.

**Pull request merged**